### ISSUE TITLE
[bug] SHORT DESCRIPTION

### ISSUE THREAD

**Issue by martinrotter** [conan-io/conan-center-index#9871]
### Environment Details (include every applicable attribute)
  * Operating System+version: win10
  * Compiler+version: msvc16
  * Conan version: 1.46
  * Python version: 3

### Steps to reproduce (Include if Applicable)

Download qt6.2.3 recipe to local folder and run "conan build .".

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
C:\aa>conan build .
WARN: Remotes registry file missing, creating default one in C:\Users\rotter\.conan\remotes.json
Auto detecting your dev setup to initialize the default profile (C:\Users\rotter\.conan\profiles\default)
Found Visual Studio 17
Found gcc 8.3
Default settings
        os=Windows
        os_build=Windows
        arch=x86_64
        arch_build=x86_64
        compiler=Visual Studio
        compiler.version=17
        build_type=Release
*** You can change them in C:\Users\rotter\.conan\profiles\default ***
*** Or override with -s compiler='other' -s ...s***


ERROR: conanfile.py (qt/None): Error in config_options() method, line 195
        if m not in self._get_module_tree:
while calling '_get_module_tree', line 157
        assert config.sections()
        AssertionError:
```

**Comment by ericLemanissier** [comment]
`conan build` command does not seem to have a way to specify the reference your want to build. this means that the recipe does not know which version you want to build. I don't think this command can be used to build any package on Conan-Center-Index

**Comment by martinrotter** [comment]
> `conan build` command does not seem to have a way to specify the reference your want to build. this means that the recipe does not know which version you want to build. I don't think this command can be used to build any package on Conan-Center-Index

I see, I likely misunderstood it all.

Is there a recommended way of building conan-center packages locally? Ideally also build all dependecies locally?

**Comment by ericLemanissier** [comment]
If you simply need to use qt, you just add it to the requirements of your conanfile.txt or conanfile.py (for example see step 4. at https://docs.conan.io/en/latest/getting_started.html#an-md5-hash-calculator-using-the-poco-libraries) and the you can call `conan install` on your conanfile.txt/conanfile.py as in step 5

**Comment by intelligide** [comment]
> Is there a recommended way of building conan-center packages locally? Ideally also build all dependecies locally?

`conan install qt/<version>@ --build`

**Comment by martinrotter** [comment]
> If you simply need to use qt, you just add it to the requirements of your conanfile.txt or conanfile.py (for example see step 4. at https://docs.conan.io/en/latest/getting_started.html#an-md5-hash-calculator-using-the-poco-libraries) and the you can call `conan install` on your conanfile.txt/conanfile.py as in step 5

No, I want to build Qt myself for very specific reasons. At this point I use custom crafter powershell scripts to do that, but thought about trying conan as it seems to take care of building dependencies etc.
